src/routes/editor/[bookID]/+page.svelte

```diff
@@ -250,22 +322,82 @@
 			{ align: 'right' },
 			{ align: 'justify' },
 			'comments-add',
-			'comments-toggle'
-		]
+			'comments-toggle',
+			'illustrations-add',
+		],
 	];
 
 	$: commentBgColor = showComments ? 'var(--color-primary-500)' : '';
-	$: cssVarStyles = `--comment-bg-color:${commentBgColor}`;
+	$: illustrationBgColor = showIllustrations ? 'var(--color-warning-800)' : '';
+	$: cssVarStyles = `--comment-bg-color:${commentBgColor}; --illustration-bg-color:${illustrationBgColor};`;
 
+	/**
+	 * Toggles the showComments boolean and updates the quill module
+	 *
+	 */
 	function toggleShowComments() {
+		if (showIllustrations && !showComments) {
+			toggleShowIllustrations();
+		}
 		showComments = !showComments;
 	}
 
+	/**
+	 * Toggles the showIllustrations boolean and updates the quill module
+	 */
+	function toggleShowIllustrations() {
+		if (showComments && !showIllustrations) {
+			toggleShowComments();
+		}
+		showIllustrations = !showIllustrations;
+		// if (showIllustrations) quill.getModule('quillIllustration').addIllustrationStyle('green')
+		// else quill.getModule('quillIllustration').addIllustrationStyle('transparent')
+	}
+
 	function removeComment(id: string) {
 		let editor = document.getElementById('editor');
 		quill.removeComment(id, editor);
 	}
 
+	/**
+	 * Removes the illustration from the quill Parchment
+	 * Also removes the illustration from supabase storage
+	 * @param id
+	 */
+	function removeIllustration(id: string) {
```

Comment:
Could you please move this logic to quill.ts. I would like the logic that is directly related to the quill editor to be there so that we avoid making this svelte class too big.